### PR TITLE
Songs model

### LINKED ISSUE #8: Song model

### PR DESCRIPTION
**Pull request by LatidoReMe**
close #8 

+ Created song model (shocking, I know)
+ Did some minor magicollection work (including making a countdown, woo-hoo)
+ Did some minor forms work (for countdown, mainly)

### CODE REVIEW

File: sukutomo/models.py [hunk 1]
```diff
@@ -310,3 +310,122 @@ def get_status(self, version='JP'):
...
+        ('hard', 'HARD'),
+        ('expert', 'EXPERT'),
+        ('master', 'MASTER'),
+    )
+
+    SONGWRITERS_DETAILS = [
+        ('composer', _('Composer')),
+        ('lyricist', _('Lyricist')),
+        ('arranger', _('Arranger')),
+    ]
```
Comment (db0company) [inline sukutomo/models.py]: I think you accidentally duplicated that, it's also down there.

File: sukutomo/models.py [hunk 2]
```diff
@@ -310,3 +310,122 @@ def get_status(self, version='JP'):
...
+    c_versions = models.TextField(_('Server availability'), blank=True, null=True, default='"JP"')
+
+    LOCATIONS_CHOICES = [
+        ('hits', _('Hits')),
+        ('daily', _('Daily rotation')),
+        ('bside', _('B-Side')),
+    ]
+    c_locations = models.TextField(_('Locations'), blank=True, null=True)
+    
+    unlock = models.PositiveIntegerField(_('Unlock'), help_text=_('Will be displayed as "Rank __"'), null=True)
```
Comment (db0company) [inline sukutomo/models.py]: This help text is for internal use only so don't translate it.

File: sukutomo/magicollections.py [hunk 3]
```diff
@@ -312,3 +312,84 @@ class EditView(MagiCollection.EditView):
...
+            if exclude_fields is None: exclude_fields = []
+            if order is None: order = []
+            status = getattr(item, 'status')
+            if status and status != 'ended':
+                start_date = getattr(item, 'b_side_start')
+                end_date = getattr(item, 'b_side_end')
+                extra_fields += [
+                    ('countdown', {
+                        'verbose_name':  _('Countdown'),
+                        'value': mark_safe(u'<span class="fontx1-5 countdown" data-date="{date}" data-format="{sentence}"></h4>').format(
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/magicollections.py]: This was written before you had implemented toCountDown(), I believe. I will definitely work on making this change when I can!

File: sukutomo/models.py [hunk 4]
```diff
@@ -310,3 +310,122 @@ def get_status(self, version='JP'):
...
+        release_date = getattr(self, 'release')
+        b_side = getattr(self, 'status')
+        if b_side is 'current':
+            return 'currently available'
+        elif timezone.now() >= release_date:
+            return 'currently available'
+        else:
+            return 'not available'
+
+    available = property(lambda _s: _s.get_availability())
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/models.py]: It's used internally, and in a later commit I did alter it to use True and False! : https://github.com/MagiCircles/SchoolIdolTomodachi/blob/CardSetSkill-Form/sukutomo/models.py#L376-L385

File: sukutomo/models.py (hunk 1 above)
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/models.py (hunk 2 above)
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/migrations/0007_auto_20180623_2152.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            field=models.TextField(null=True, verbose_name='Favorite food'),
+            preserve_default=True,
+        ),
+        migrations.AlterField(
+            model_name='idol',
+            name='i_year',
+            field=models.PositiveIntegerField(null=True, verbose_name='School year', choices=[(0, '1st year'), (1, '2nd year'), (2, '3rd year')]),
+            preserve_default=True,
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/migrations/0007_auto_20180623_2152.py]: ✅

**Comment by LatidoReMe** [comment]
Moved this into: #40

**Pull request closed without merging**